libs/wingsdk/src/platform/platform.ts

```diff
@@ -0,0 +1,45 @@
+import { Construct } from "constructs";
+import { App } from "../core";
+
+/**
+ * Platform interface
+ */
+export interface IPlatform {
+  /**
+   * The model the platform is built for
+   *
+   * @example "sim"
+   */
+  readonly model: string;
+
+  // TODO: implement newInstance
+  // newInstance?(type: string, scope: Construct, id: string, props: any): any;
```

Comment:
Theres some additional complexity to get this API to work, due to the rapid development on Wing and the sheer size of the required changes for platforms I want to introduce this initial implementation so I can get out of rebase HELL